blocks/bricks/sequence_generators.py

```diff
@@ -162,7 +162,7 @@ def cost(self, application_call, outputs, mask=None, **kwargs):
 
         # Separate the deliverables
         states = {name: results[name][:-1] for name in self.state_names}
-        glimpses = {name: results[name] for name in self.glimpse_names}
+        glimpses = {name: results[name][1:] for name in self.glimpse_names}
```

Comment:
As for me, it needs more explanation. Why are you not using the last state and the first glimpse?